lib/msf/base/config.rb

```diff
@@ -406,6 +411,11 @@ def user_script_directory
     config_directory + FileSep + "scripts"
   end
 
+  # @return [String] path to user-specific data directory.
+  def user_data_directory
+    config_directory + FileSep + self['DataDirectory']
```

Comment:
This doesn't seem to be user configurable and tbh I'm not sure that this is all that random.

As noted in the function description it simply:

```
  # Returns the configuration directory default.
  #
  # @return [String] the root configuration directory.
  def config_directory
```

There is no mention that this is user-specific and its very possible that multiple users could share the same root configuration directory. If you are going to implement this I'd be in favor of something that maybe grabs the current username and uses that to create a unique directory somewhere to make things more user-specific if that is what you are going for.

Reply:
> There is no mention that this is user-specific and its very possible that multiple users could share the same root configuration directory.

I believe the semantics of locating the user's `config_directory` is defined here:

https://github.com/rapid7/metasploit-framework/blob/e445a39719e03b54fb0a32a9e0b5c07d74e505b7/lib/msf/base/config.rb#L25-L55

It can be influenced by the `MSF_CFGROOT_CONFIG` environment variable, but will generally end up resolving to `~/.msf4` i.e. the user's home directory, which I believe covers what you were requesting 👀

Reply:
The intent was to keep parity between the install dir and a user's config dir. It's just a way to reference `~/.msf4/data` so people can keep their private module dependencies portable between computers that may have different user accounts.

I don't know if it's necessary to introduce I'll `sudo -Es msfconsole` so I keep the low priv user's shell environment, so ~/.msf4 isn't in /root.

Reply:
Woops appears that I misread the code, my apologies for that, not sure how I got to the conclusion I did last night but agree with your logic.

As for the `sudo -Es msfconsole`, sounds good :)